tools/tslint-rules/coercionTypesRule.ts

```diff
@@ -38,6 +47,48 @@ class Walker extends Lint.RuleWalker {
     super.visitClassDeclaration(node);
   }
 
+  /**
+   * Checks if the given property declaration of a coercion member includes null and
+   * undefined in the type node. We enforce that all acceptance members accept these
+   * values since we want coercion inputs to work with the async pipe.
+   */
+  private _lintCoercionMember(node: ts.PropertyDeclaration) {
+    if (!node.type) {
+      this.addFailureAtNode(node, 'Acceptance member needs to have an explicit type.');
+      return;
+    }
+
+    if (node.type.kind === ts.SyntaxKind.AnyKeyword) {
+      // if the type is "any", then it can be "null" and "undefined" too.
+      return;
```

Comment:
Do we want to allow it to be `any`? Since it's something we control it might make sense to be a bit more strict.

Reply:
Some coercion members are set to `any`. i.e. the value input for mat-input's because otherwise `value` would only accept string while datepicker input accepts a date too.

So `any` is a special case that we allow.